### PR TITLE
Added OpenGenerics feature to the IoC

### PR DESCRIPTION
**Pull request by fedemkr**
### :sparkles: What kind of change does this PR introduce? (Bug fix, feature, docs update...)
This introduces OpenGenerics feature to the IoC. Now we can register things like IMyInterface<> without specifying the generic type parameter that will be set at the moment of resolving it.
### :arrow_heading_down: What is the current behavior?
It does not allow to use open generics in the IoC.
### :boom: Does this PR introduce a breaking change?
No.
### :memo: Links to relevant issues/docs
One of the improvements to do in [#2069](https://github.com/MvvmCross/MvvmCross/issues/2069)
### :thinking: Checklist before submitting

- [x] All projects build
- [x] Follows style guide lines ([code style guide](https://github.com/MvvmCross/MvvmCross#code-style-guidelines))
- [ ] Relevant documentation was updated ([docs style guide](https://www.mvvmcross.com/documentation/contribute/mvvmcross-docs-style-guide))
- [ ] Nuspec files were updated (when applicable)
- [ ] Rebased onto current develop

### CODE REVIEW

File: MvvmCross/Platform/Platform/IoC/MvxSimpleIoCContainer.cs [hunk 1]
```diff
@@ -268,7 +294,12 @@ public void RegisterType(Type t, Func<object> constructor)
 
         public void RegisterType(Type tInterface, Type tConstruct)
```
Comment (Cheesebaron) [inline MvvmCross/Platform/Platform/IoC/MvxSimpleIoCContainer.cs]: rename to `interfaceType` and `constructType`

File: MvvmCross/Platform/Platform/IoC/MvxSimpleIoCContainer.cs [hunk 2]
```diff
@@ -364,6 +395,13 @@ public void CallbackWhenRegistered(Type type, Action action)
             action();
         }
 
+        public void CleanAllResolvers()
+        {
+            this._resolvers.Clear();
```
Comment (Cheesebaron) [inline MvvmCross/Platform/Platform/IoC/MvxSimpleIoCContainer.cs]: remove `.this`

File: MvvmCross/Platform/Platform/IoC/MvxSimpleIoCContainer.cs [hunk 3]
```diff
@@ -388,7 +426,7 @@ private bool InternalTryResolve(Type type, out object resolved)
         private bool InternalTryResolve(Type type, ResolverType? requiredResolverType, out object resolved)
         {
             IResolver resolver;
-            if (!_resolvers.TryGetValue(type, out resolver))
+            if (!this.TryGetResolver(type, out resolver))
```
Comment (Cheesebaron) [inline MvvmCross/Platform/Platform/IoC/MvxSimpleIoCContainer.cs]: remove `.this`

**Comment by martijn00** [comment]
Can you add your email address that you use for commits to your account?

**Comment by MarcBruins** [comment]
@fedemkr can you add this to the docs: https://www.mvvmcross.com/documentation/fundamentals/inversion-of-control-ioc?

**Pull request merged**